[Dynamic] Fix dynamic shape size inspection bug

oulgen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120355
* __->__ #120341

Fixes #120198


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

Differential Revision: [D54035984](https://our.internmc.facebook.com/intern/diff/D54035984)

oulgen committed:
[Dynamic] Fix dynamic shape size inspection bug

[ghstack-poisoned]

oulgen committed:
Update on "[Dynamic] Fix dynamic shape size inspection bug"


Fixes #120198


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54035984).

ezyang reviewed (approved):
thanks, much appreciated. If you're down for it, also looking into why this didn't just cause a graph break would also be interesting

ezyang commented on test/dynamo/test_triton_kernels.py:
```diff
@@ -924,6 +924,40 @@ def f(x, y):
+
+        x = torch.randn(4, device="cuda")
+        eager_out = f(x)
+        compiled_out = torch.compile(f, backend=backend)(x)
```
consider fullgraph=True here

ezyang commented on test/dynamo/test_triton_kernels.py:
```diff
@@ -924,6 +924,40 @@ def f(x, y):
+    @torch._dynamo.config.patch(capture_dynamic_output_shape_ops=True)
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
+    @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
+    def test_triton_kernel_dynamic_shape_tensor(self, backend):
```
I would actually suggest `unbacked` rather than `dynamic` in the name here, that's what we've been using to identify tests that involve data dependent; it disambiguates between u0 and s0.

oulgen committed:
Update on "[Dynamic] Fix dynamic shape size inspection bug"


Fixes #120198


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

Differential Revision: [D54035984](https://our.internmc.facebook.com/intern/diff/D54035984)

[ghstack-poisoned]

oulgen commented:
@oulgen has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54035984).

oulgen commented:
@pytorchbot merge -f "flaky test"
